codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -818,6 +818,10 @@ public ChannelFuture goAway(final ChannelHandlerContext ctx, final int lastStrea
             return promise;
         }
 
+        // The GO_AWAY may be the result of an internal error, or user generated (and they may not know the stream ID).
+        // We fire this event in the pipeline so the application has visibility that the connection is shutting down.
+        ctx.fireUserEventTriggered(new Http2GoAwaySentEvent(lastStreamId, errorCode));
```

Comment:
So we are saying it is generally useful right ? Not only for the child channel / frame API. I think this makes sense even when in the commit message it really sounds like it may be only useful for people that use the child channel. API. Maybe we should clarify it a bit in the commit message.

Reply:
Yes it is generally useful everywhere. Mentioning the child channel API was intended to provide motivation for the approach. For the non-child channel API we may be able to have methods overridable (which is what we have done in the past), but the user event can be consumed in both cases. I can clarify the commit message.